### PR TITLE
[TechDocs] Beta!

### LINKED ISSUE #4367: TechDocs: Use lowercase entity kind, namespace and name in URLs
## Feature Suggestion

When you open a TechDocs page like https://demo.backstage.io/docs/default/Component/backstage

The entity namespace (e.g. `default`) and kind (e.g. `Component`) and name should be in lowercase.

## Possible Implementation

Look at `/catalog` plugin for inspiration.

https://github.com/backstage/backstage/blob/8e1c7be11c916d95e9281914b98c59a79d0a1ada/plugins/catalog-react/src/routes.ts#L39-L46

### LINKED ISSUE #6778: [TechDocs] Finish composability API migration
## 🧹  What/Why

In #6458, we (optionally) split up `/docs` and `/docs/:namespace/:kind/:name/*` into distinct routes in order to enable composability/customization of the TechDocs Home layout.

This presents a good opportunity to wrap-up the migration to the Composability API.  I think we can call things complete if we...

- [ ] Replace uses of `react-router-dom`'s `generatePath` to link to docs pages with the [composability API](https://backstage.io/docs/plugins/composability#parameterized-routes) equivalent
- [ ] Replace hard-coded `/docs/...` links throughout Backstage code, e.g. [here](https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/AboutCard/AboutCard.tsx#L110), which may be the only one?

### PR DESCRIPTION
**Pull request by camilaibs**
## What / Why

We're wrapping up the [TechDocs Beta Milestone](https://github.com/backstage/backstage/milestone/29)!  And the last remaining issues touch upon some similar bits of code and have breaking changes, so we've bundled them together.

### Allow lowercase entity triplets when accessing TechDocs Sites
Allows access to TechDocs content without needing entity details to be case-sensitive (e.g. docs for entities of type `Component` can be accessed with a url like `https://backstage.example.net/docs/default/component/some-software/getting-started`)

__Breaking change__: No change for those with `techdocs.builder` set to `local`.  For those who build docs in CI/CD: a migration must be run prior to going live (because objects must be moved in underlying storage providers).

An escape hatch is provided to integrators who run into problems with the migration process in the form of a boolean `techdocs.legacyUseCaseSensitiveTripletPaths` configuration, which can be set to revert to the old behavior/expectations.

Fixes #4367

### Finish composability API migration
Enables folks to choose their own adventure when mounting TechDocs (rather than hard-coding assumptions around TechDocs living at`/docs/...`).

__Breaking change__: App integrators must mount `<TechDocsReaderPage />` to a path (like `/docs/:namespace/:kind/:name/*`).

Fixes #6778 

------

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by emmaindal — APPROVED** [review]
⭐  Awesome to see the additional fall back configuration, migration scripts and migration documentation for a smooth migration journey! Well done!

File: docs/features/techdocs/how-to-guides.md [hunk 1]
```diff
@@ -104,83 +104,110 @@ the repository. The archive does not have any git history attached to it. Also
...
+
+> This guide only applies to the "recommended" TechDocs deployment method (where
+> an external storage provider and external CI/CD is used). If you use the
+> "basic" or "out-of-the-box" setup, you can stop here! No action needed.
+
+The beta version of TechDocs (v0.x.y) made a breaking change to the way TechDocs
+content was accessed and stored, allowing pages to be accessed with
+case-insensitive entity triplet paths (e.g. `/docs/namespace/kind/name` whereas
+in prior versions, they could only be accessed at `/docs/namespace/Kind/name`).
+In order to enable this change, documentation has be stored in an external
```
Comment (soapraj) [inline docs/features/techdocs/how-to-guides.md:155]:
```suggestion
In order to enable this change, documentation has to be stored in an external
```

**Comment by camilaibs** [comment]
@backstage/reviewers looking for one more approval 🙏🏻

File: .changeset/catalog-compose-my-techdocs.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/plugin-catalog': patch
```
Comment (Rugvip) [inline .changeset/catalog-compose-my-techdocs.md:2]: Bit torn on whether we can actually call this patch. It's definitely gonna break behaviour, but not compilation or runtime. Perhaps we can mark it a bit more visibly with a **NOTE** or mth?

**Review by Rugvip — APPROVED** [review]
Noice 👌

File: .changeset/catalog-compose-my-techdocs.md (hunk 2 above)
Comment (iamEAP) [reply .changeset/catalog-compose-my-techdocs.md:2]: We had a similar pause when creating this, yeah.  Are you thinking literally just a bold **Note** in the changeset text @Rugvip?

**Pull request merged**